util/jsonrpc-types/src/blockchain.rs

```diff
@@ -119,53 +119,62 @@ impl From<CellOutput> for CoreCellOutput {
 }
 
 #[derive(Clone, Default, Serialize, Deserialize, PartialEq, Eq, Hash, Debug)]
-pub struct CellOutPoint {
+pub struct OutPoint {
     pub tx_hash: H256,
     pub index: Unsigned,
 }
 
-impl From<CoreCellOutPoint> for CellOutPoint {
-    fn from(core: CoreCellOutPoint) -> CellOutPoint {
+impl From<CoreOutPoint> for OutPoint {
+    fn from(core: CoreOutPoint) -> OutPoint {
         let (tx_hash, index) = core.destruct();
-        CellOutPoint {
+        OutPoint {
             tx_hash,
             index: Unsigned(u64::from(index)),
         }
     }
 }
 
-impl From<CellOutPoint> for CoreCellOutPoint {
-    fn from(json: CellOutPoint) -> Self {
-        let CellOutPoint { tx_hash, index } = json;
-        CoreCellOutPoint {
+impl From<OutPoint> for CoreOutPoint {
+    fn from(json: OutPoint) -> Self {
+        let OutPoint { tx_hash, index } = json;
+        CoreOutPoint {
             tx_hash,
             index: index.0 as u32,
         }
     }
 }
 
-#[derive(Clone, Default, Serialize, Deserialize, PartialEq, Eq, Hash, Debug)]
-pub struct OutPoint {
-    pub cell: Option<CellOutPoint>,
-    pub block_hash: Option<H256>,
+#[derive(Clone, Serialize, Deserialize, PartialEq, Eq, Hash, Debug)]
+#[serde(rename_all = "snake_case")]
```

Comment:
It's better to use `Internally tagged` enum representation.

Reply:
What does the JSON looks like of this type?

I recommend to use following structure:

```json
{ "type": "cell", "previous_output": { "index": "0", "tx_hash": "0x04e427..." } }
{ "type": "cell_with_header", "previous_output": { "index": "0", "tx_hash": "0x04e427..." }, "header_hash": "0xabcd..." }
{ "type": "dep_group", "previous_output": { "index": "0", "tx_hash": "0x04e427..." } }
{ "type": "header", "header_hash": "0xabcd..." }
```

A simple way to achieve this is to modify the enum structure to

```rust
#[serde(rename_all = "snake_case", tag = "type")]
pub enum CellDep {
  Cell { previous_output: OutPoint },
  CellWithHeader { previous_output: OutPoint, header_hash: H256 },
  DepGroup { previous_output: OutPoint },
  Header { header_hash: H256 },
}
```

Also it's better to include 4 different dep types in the RPC doc example response.

Reply:
Since `ScriptHashType` is Camel-Case, can we keep the type field the default way?
``` javascript
{ "type": "Cell", "previous_output": { "index": "0", "tx_hash": "0x04e427..." } }
{ "type": "CellWithHeader", "previous_output": { "index": "0", "tx_hash": "0x04e427..." }, "header_hash": "0xabcd..." }
{ "type": "DepGroup", "previous_output": { "index": "0", "tx_hash": "0x04e427..." } }
{ "type": "Header", "header_hash": "0xabcd..." }
```

Reply:
Or change `ScriptHashType` to `snake_case` ?